python/cudf/cudf/core/index.py

```diff
@@ -1212,37 +1161,63 @@ def get_loc(self, key, method=None, tolerance=None):
                 "is specified."
             )
 
-        key_as_table = cudf.core.frame.Frame(
-            {"None": as_column(key, length=1)}
+        needle_table = cudf.DataFrame(
+            {"None": as_column(target), "order": arange(0, len(target))}
+        )
+        haystack_table = cudf.DataFrame(
+            {"None": self._column, "order": arange(0, len(self))}
         )
+        merged_table = haystack_table.merge(
+            needle_table, on="None", how="outer"
+        )
+        result_series = (
+            merged_table.sort_values(by="order_y")
+            .head(len(target))["order_x"]
+            .reset_index(drop=True)
+        )
```

Comment:
This could be constructed with an inner join, right?
```python
needle = as_column(target)
haystack = self._column
result = cudf.core.column.full(len(needle), fill_value=-1, dtype="int32") # TODO: use size_type
scatter_map, indices = libcudf.join.join([needle], [haystack], how="inner")
(result,) = libcudf.copying.scatter([indices], scatter_map, [result])
```

Reply:
Right, made the change to a simpler inner join.